plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -17,10 +17,14 @@ async function generateReviewAssignments({
   // Get application/reviewId from applicationData if not provided in parameters
   const applicationId = parameters?.applicationId ?? applicationData?.applicationId
   const reviewId = parameters?.reviewId ?? applicationData?.reviewData?.reviewId
-  const isReview =
-    parameters?.isReview === false
-      ? false
-      : parameters?.isReview || applicationData?.action_payload?.trigger_payload?.table === 'review'
+  // Check if "isReview = false" to overwreite having received reviewId - used when need to process an upgrade on same review levels
+  const overwriteIsReview = parameters?.isReview
+
+  // Set "isReview = true" when receiving reviewId (and isReview != false) OR triggered from table 'review'
+  // Even if "isReview === true" is received, but no reviewId it will be considered "isReview = false"
+  const isReview = !!overwriteIsReview
+    ? overwriteIsReview && reviewId !== undefined
+    : reviewId !== undefined || applicationData?.action_payload?.trigger_payload?.table === 'review'
```

Comment:
Not sure I understand why this was needed, it seems more complicated to me. What is the case that this accounts for that my previous logic didn't? (Feel free to explain it to me in person if that's easier -- I'm more just interested than anything)

Reply:
It wasn't finding `isReview` on the params and returning `isReview` false, so let's act as it's a Application Submission